.github/workflows/main.yml

```diff
@@ -17,88 +17,91 @@ permissions:
+    runs-on: ubuntu-latest
     
     env:
-      # Test filter matching AppVeyor configuration
-      TEST_FILTER: "TestCategory!=localOnly&TestCategory!=Benchmark&TestCategory!=SOAP&TestCategory!=REST"
+      DOTNET_NOLOGO: true
+      DOTNET_CLI_TELEMETRY_OPTOUT: true
 
     steps:
     - name: Checkout
       uses: actions/checkout@v4
       with:
         fetch-depth: 0
 
-    - name: Setup MSBuild
-      uses: microsoft/setup-msbuild@v2
-
-    - name: Setup NuGet
-      uses: nuget/setup-nuget@v2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
       with:
-        nuget-version: 'latest'
-
-    - name: Setup VSTest
-      uses: darenm/Setup-VSTest@v1.3
-
-    - name: Restore NuGet packages
-      run: |
-        # First restore using nuget.exe which handles packages.config correctly
-        nuget restore Qwiq.sln -NonInteractive -PackagesDirectory packages -ConfigFile nuget.config
-        
-        # Also restore each packages.config individually to ensure all dependencies are resolved
-        Get-ChildItem -Path . -Recurse -Filter packages.config | ForEach-Object {
-          Write-Host "Restoring packages from: $($_.FullName)"
-          nuget restore $_.FullName -PackagesDirectory packages -ConfigFile nuget.config
-        }
-      shell: pwsh
-      continue-on-error: false
-
-    - name: Build solution
-      run: msbuild Qwiq.sln /p:Configuration=Release /p:Platform="Any CPU" /v:minimal /m
-
-    - name: Run tests
-      shell: pwsh
-      run: |
-        # Find all test assemblies (excluding IntegrationTests)
-        $testAssemblies = @(
-          Get-ChildItem -Path "test" -Recurse -Filter "*Tests.dll" | 
-          Where-Object { 
-            $_.FullName -match "bin\\Release" -and 
-            $_.FullName -notmatch "IntegrationTests" -and
-            $_.FullName -notmatch "\\ref\\" 
-          }
-        )
-        
-        if ($testAssemblies.Count -eq 0) {
-          Write-Warning "No test assemblies found!"
-          exit 1
-        }
-        
-        Write-Host "Found $($testAssemblies.Count) test assemblies"
-        
-        foreach ($assembly in $testAssemblies) {
-          Write-Host "Testing: $($assembly.FullName)"
-          vstest.console.exe $assembly.FullName /Logger:trx /TestCaseFilter:"$env:TEST_FILTER"
-          if ($LASTEXITCODE -ne 0) {
-            Write-Warning "Tests failed in $($assembly.Name) with exit code $LASTEXITCODE"
-          }
-        }
+        dotnet-version: '8.0.x'
+
+    - name: Restore packages
+      run: dotnet restore Qwiq.sln
+
+    - name: Build solution (net8.0 only on Ubuntu)
+      run: dotnet build Qwiq.sln --configuration Release --no-restore --framework net8.0
+
+    - name: Run tests (net8.0)
+      run: dotnet test Qwiq.sln --configuration Release --no-build --framework net8.0 --logger trx --results-directory TestResults --filter "TestCategory!=localOnly&TestCategory!=Benchmark&TestCategory!=SOAP&TestCategory!=REST&TestCategory!=IntegrationTests"
```

Comment:
the filter can be extracted to an environment variable and reused, so it's consistent between the two jobs